guide/src/async-await.md

```diff
@@ -30,8 +30,7 @@ Resulting future of an `async fn` decorated by `#[pyfunction]` must be `Send + '
 
 As a consequence, `async fn` parameters and return types must also be `Send + 'static`, so it is not possible to have a signature like `async fn does_not_compile(arg: &PyAny, py: Python<'_>) -> &PyAny`.
 
-It also means that methods cannot use `&self`/`&mut self`, *but this restriction should be dropped in the future.*
-
+However, there is an exception for method receiver, so async methods can accept `&self`/`&mut self`
```

Comment:
Is it worth pointing out here that there's a good case for being extra careful with `&mut self` as a method receiver in a future? I think it's very likely users will accidentally hit borrow errors due to overlapping futures.

We could potentially restrict this to `&self` on frozen classes? That would help prevent users creating aliasing borrow errors from async code. What do you think about that restriction?

Reply:
> We could potentially restrict this to &self on frozen classes?

Honestly, I prefer to just add a warning in the documentation. I don't want to prevent careful users to use a `mpsc::Receiver` in an efficient way. 
Maybe you could ping another maintainer to discuss this decision?

Reply:
Ok let's keep this. I think what I'd prefer to do is make frozen the default and then document what the risks of `&mut` are if users choose to enable this.